### PR TITLE
chore(errorHandler):new abortWithStatus method with Json body

### PR DESCRIPTION
**Pull request by pjgg**
chore(errorHandler):new abortWithStatus method with Json body and contentType application/json

### CODE REVIEW

File: context.go [hunk 1]
```diff
@@ -119,6 +119,13 @@ func (c *Context) AbortWithStatus(code int) {
 	c.Abort()
 }
 
+// AbortWithStatusJson calls `Abort()` and then `JSON` internally. This method stops the chain, writes the status code and return a JSON body
+// It also sets the Content-Type as "application/json".
+func (c *Context) AbortWithStatusJson(code int, obj interface{}) {
```
Comment (appleboy) [inline context.go]: I think `AbortWithStatusJSON` is better than `AbortWithStatusJson`

**Comment by appleboy** [comment]
@tboerger Please help to review.

**Comment by javierprovecho** [comment]
@appleboy switching target branch to `develop`

File: context_test.go [hunk 2]
```diff
@@ -595,6 +595,17 @@ func TestContextAbortWithStatus(t *testing.T) {
...
+func TestContextAbortWithStatusJSON(t *testing.T) {
+	c, w, _ := CreateTestContext()
+	c.index = 4
+	c.AbortWithStatusJSON(415, "{\"foo\":\"bar\", \"barTwo\":\"fooTwo\"}")
+
+	assert.Equal(t, c.index, abortIndex)
+	assert.Equal(t, c.Writer.Status(), 415)
+	assert.Equal(t, w.Code, 415)
+	assert.True(t, c.IsAborted())
+}
```
Comment (javierprovecho) [inline context_test.go]: It is required to also test the output of that function, by checking `body == json{...}` and if `content_type == application/json`. Else looks good to me. Thanks @pjgg 

cc @appleboy

Comment (appleboy) [inline context_test.go]: @javierprovecho Good catch.

**Comment by appleboy** [comment]
@pjgg conflicts.

You need to create new branch base on `develop` not `master`.

File: context_test.go [hunk 3]
```diff
@@ -15,6 +15,8 @@ import (
 	"testing"
 	"time"
```
Comment (appleboy) [inline context_test.go]: remove empty line.

File: context_test.go [hunk 4]
```diff
@@ -15,7 +15,8 @@ import (
 	"strings"
 	"testing"
 	"time"
-
+	"fmt"
+	"github.com/manucorporat/sse"
```
Comment (appleboy) [inline context_test.go]: change to `gopkg.in/gin-contrib/sse.v0`

File: context_test.go [hunk 5]
```diff
@@ -15,7 +15,8 @@ import (
 	"strings"
 	"testing"
 	"time"
-
+	"fmt"
```
Comment (appleboy) [inline context_test.go]: For imports you should use the following format (without the comments)

```go
import (
  // stdlib
  "encoding/json"
  "fmt"

  // external packages
  "github.com/foo/bar"
  "gopkg.io/baz.v1"
)
```

Comment (appleboy) [inline context_test.go]: build fails. Duplicated import `fmt`

**Comment by appleboy** [comment]
@pjgg Thanks.

**Pull request merged**